src/Type/InternalType.php

```diff
@@ -0,0 +1,286 @@
+
+            /*
+             * Text types.
+             */
+
+            case 'bpchar': // PostgreSQL
+            case 'char':
+            case 'character':
+            case 'nchar': // SQL Server
+                return InternalType::CHAR;
+
+            case 'ntext': // SQL Server
+            case 'string':
+            case 'text':
+                return InternalType::TEXT;
+
+            case 'char varying':
+            case 'character varying':
+            case 'nvarchar': // SQL Server
+            case 'varchar':
+            case 'varchar2': // Oracle
+            case 'varying character': // SQLite
+                return InternalType::VARCHAR;
+
+            /*
+             * Binary types.
+             */
+
+            case 'binary': // SQL Server
+            case 'blob':
+            case 'bytea': // PostgreSQL
+            case 'varbinary':
+                return InternalType::BINARY;
+
+            /*
+             * Date and time.
+             */
+
+            case 'interval':
+                return InternalType::DATE_INTERVAL;
+
+            case 'date':
+                return InternalType::DATE;
+
+            case 'time': // PostgreSQL
+                return InternalType::TIME;
+
+            case 'timez':
+                return InternalType::TIME;
+
+            case 'datetime':
+            case 'datetime2': // SQL Server
+            case 'datetimez': // PostgreSQL
+            case 'timestamp':
+            case 'timestampz': // PostgreSQL
+                return InternalType::TIMESTAMP;
+
+            /*
+             * Specific types.
+             */
+
+            case 'json':
+            case 'jsonb': // PostgreSQL
+                return InternalType::JSON;
+
+            case 'guid':
+            case 'uniqueidentifier':
+            case 'uuid':
+                return InternalType::UUID;
+
+            /*
+             * Unknown type.
+             */
+
+            default:
+                return InternalType::UNKNOWN;
+        }
+    }
+
+    public function precisionIsRelevant(): bool
+    {
+        return $this === self::DECIMAL;
+    }
+
+    public function scaleIsRelevant(): bool
+    {
+        return $this === self::DECIMAL;
+    }
+
+    public function lengthIsRelevant(): bool
+    {
+        return $this === self::CHAR || $this === self::VARCHAR;
+    }
+
+    public function isDate(): bool
+    {
+        return $this === self::DATE || $this === self::TIMESTAMP;
+    }
+
+    public function isNumeric(): bool
+    {
+        return
+            $this === self::DECIMAL ||
+            $this === self::FLOAT || $this === self::FLOAT_BIG || $this === self::FLOAT_SMALL ||
+            $this === self::INT || $this === self::INT_BIG || $this === self::INT_SMALL ||
+            $this === self::IDENTITY || $this === self::IDENTITY_BIG || $this === self::IDENTITY_SMALL ||
+            $this === self::SERIAL || $this === self::SERIAL_BIG || $this === self::SERIAL_SMALL
+        ;
```

Comment:
Would you not prefer to use `\in_array()` ?

Reply:
Really not, I'm afraid `\in_array()` would be slower, although now that I have benchmarks, I could test it!